File: php/bench.php
<?php

namespace banch;

function execute($target, $count)
{
    require_once "script/$target.php";
    $func = "$target\\main";

    $start = microtime(true);

    for ($i=0; $i < $count; ++$i) {
        $func();
    }

    return microtime(true) - $start;
}

$count = $argv[1];
$targetList = array_slice($argv, 2);

$blank = execute('blank', $count);

foreach ($targetList as $target) {
    $time = execute($target, $count) - $blank;

    print "# $target\n";
    print "## $time\n";
}

File: perl/bench.pl
package bench;
use 5.16.0;
use Module::Load;
use Time::HiRes qw(gettimeofday);

sub execute {
    my ($target, $count) = @_;

    load $target;
    my $start = gettimeofday;
    for (0..$count - 1) {
        $target->main();
    }
    gettimeofday - $start;
}

my $count = $ARGV[0];
my @target_list = @ARGV[1..$#ARGV];
my $blank = execute('blank', $count);

foreach my $target (@target_list) {
    my $time = execute($target, $count) - $blank;
    say "# $target";
    say "## $time";
}
